### DISCUSSION TITLE
Welcome to Ktorfit Discussions!

### DISCUSSION THREAD

**Discussion by Foso** [Foso/Ktorfit#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mattrob33** [comment]
Hello! I'm glad to see this project. I've thought about making a similar one myself before. I have a couple questions after looking through the source briefly.

1. Does Ktorfit not yet support url path variables? Something like,

```Kotlin
@Get("users/{id}/")
suspend fun getUser(@Path("id") id: Int): User
```

I know this is a brand new project, so I don't mean for that to sound critical - just wondering if I missed it in the source.

2. Why did you decide to manually implement source generation instead of using KotlinPoet? Personally I'd be a lot more inclined to contribute if it were using KotlinPoet, and I'd imagine many others would feel similarly. The MyClass, MyFunction, etc., stuff feels really awkward and unclear, and it requires learning another source generation tool, as well as maintaining another source generation tool. I don't have a ton of experience with KotlinPoet but I have used it before. I'd be happy to help implement it if this is something you'd be interested in.

Again, thanks for building and sharing this! I really hope to be able to contribute and to see this project grow in functionality and adoption! 👏🏼

**Comment by Foso** [comment]
Hi @mattrob33, thank you. It does support Path variables https://foso.github.io/Ktorfit/#path

2) I did it this way because it was easier for me, because i don't have much experience with KotlinPoet. Now that i know which code i need to generate i'm interested in migrating it to KotlinPoet. I understand that my own solution is complicated to understand for other people.

**Reply by mattrob33** [reply]
Ah thanks, I didn't see that! I was looking through the source on my phone while holding my son for a nap :) Looking forward to digging into it a bit more later.

Also, I've opened an issue for moving to KotlinPoet as that's probably the more appropriate place for that discussion. Thanks!
